File: list3/java/task2/Hexagon.java
import java.lang.Math;

public final class Hexagon extends OneParameter
{
    public Hexagon(final double s) throws IllegalArgumentException
    {
        if(s<0)
        {
            throw new IllegalArgumentException("Side has to be at least 0, provided: "+s);
        }
        parameter_one = s;
    }

    public double area()
    {
        return (Math.sqrt(3) * parameter_one) * parameter_one * 3;
    }

    public double lenght()
    {
        return 6 * parameter_one;
    }

    public String name()
    {
        return "Hexagon";
    }
}

File: list3/cpp/source/hexagon.cpp
#include <hexagon.hpp>

#include <cmath>

Hexagon::Hexagon(const double input1);
:var1(input1)
{
}

const double Hexagon::area()
{
    return ((sqrt(3)*var1)*var1*3);
}

const double Hexagon::length()
{
    return (6*var1);
}